File: src/types/agent.ts
export type AgentMode = "plan" | "act";

export type AgentStatus =
  | "starting"
  | "running"
  | "waiting"
  | "completed"
  | "error";

export type AgentSession = {
  id: string;
  worktree_path: string;
  mode: AgentMode;
  status: AgentStatus;
  started_at: string;
  completed_at: string | undefined;
};

export type AgentOutputPayload = {
  session_id: string;
  data: number[];
};

export type AgentStatusPayload = {
  session_id: string;
  status: AgentStatus;
};

File: src-tauri/src/models/agent.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum AgentMode {
    Plan,
    Act,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum AgentStatus {
    Starting,
    Running,
    Waiting,
    Completed,
    Error,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AgentSession {
    pub id: String,
    pub worktree_path: String,
    pub mode: AgentMode,
    pub status: AgentStatus,
    pub started_at: String,
    pub completed_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum AgentOutputType {
    Stdout,
    Stderr,
    System,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AgentOutput {
    pub id: String,
    pub session_id: String,
    pub output_type: AgentOutputType,
    pub content: String,
    pub timestamp: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AgentOutputEvent {
    pub session_id: String,
    pub data: Vec<u8>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AgentStatusEvent {
    pub session_id: String,
    pub status: AgentStatus,
}
